Запрос на слияние ветки develop в ветку master

Manikgog opened the pull request.

Manikgog committed:
Написаны методы контроллера AnimalController, сервиса AnimalService, репозитория AnimalRepo для добавления животного, добавления фотографии животного, удаления животного, обновления сведений о животном.

Manikgog committed:
Начал писать класс для тестирования AnimalController - AnimalControllerTestRestTemplateTest. Написаны методы для заполнения тестовой базы данных H2. Написан тест для проверки загрузки животного в базу данных.

Manikgog committed:
Добавлены изображения животных.

Manikgog committed:
Написаны тестовые методы для класса AnimalControllerTestRestTemplateTest:
- insertDateOfReturningAnimal_positiveTest()
- insertHumanWhoTookAnimal_negativeTest()
- insertHumanWhoTookAnimal_positiveTest()

Manikgog committed:
Написаны методы для тестирования AnimalController'a, но не все правильно работают. Не получилось сделать методы для проверки загрузки фото и его получения, также метод для постаничного вывода.

Manikgog committed:
Попробовал сделать двух этапный ввод, т.е. пользователь вводит одну команду, ему в ответ приходит меню из нескольких команд, из которых он вводит следующую команду.

Manikgog committed:
Начал писать мок-тесты AnimalService.

Manikgog committed:
Создал 2 enum Gender и AnimalType. Изменил сущность Animal - вместо AnimalType и Gender с типа String заменил на Enum. Начал переписывать тесты AnimalControllerTestRestTemplateTest.

Manikgog committed:
Merge branch 'Maksim_feature_animal_crud' of https://github.com/Manikgog/pet_nursery_bot_v2 into Maksim_feature_animal_crud

# Conflicts:
#	src/main/java/ru/pet/nursery/handler/Handler.java

Manikgog committed:
Написан контроллер AnimalController, сервис AnimalService, репозиторий AnimalRepo. Начал писать тесты.

Manikgog committed:
Написаны методы класса AnimalControllerTestRestTemplateTest для сквозного тестирования AnimalController, AnimalService, AnimalRepo. Также написаны методы класса AnimalServiceMockTest и ValidatoreUnitTest. Написана документация для методов контроллера AnimalController для swagger.

Manikgog committed:
Добавлены комментарии в классах: Validator, Animal, AnimalDTOForUserMapper, AnimalMapper.

Roman-Vetchanin committed:
Merge pull request #5 from Manikgog/Vethanin-feature-shelter

Vethanin feature shelter

Manikgog committed:
В обработчик исключений добавил три метода обработки.

Manikgog committed:
Merge remote-tracking branch 'origin/develop' into Maksim_feature_animal_crud

# Conflicts:
#	src/main/java/ru/pet/nursery/handler/Handler.java
#	src/main/java/ru/pet/nursery/web/handler/NurseryExceptionHandler.java
#	src/main/resources/application.properties

Manikgog committed:
Слияние веток Maksim_feature_animal_crud в ветку develop

Manikgog committed:
Создан контроллер VolunteerController. Добавлено поле long telegramUserId в сущность Volunteer.

Manikgog committed:
Написан класс VolunteerValidator и юнит-тесты для него в классе VolunteerValidatorUnitTest.

Roman-Vetchanin committed:
Update application.properties

Manikgog committed:
Написано несколько методов в VolunteerController, VolunteerService для загрузки объекта в базу данных и его update.

Manikgog committed:
Написал юнит-тесты для  VolunteerService начал писать тесты для VolunteerController.

Manikgog committed:
Сделаны исправления по замечаниям наставника.

Manikgog committed:
Написал тесты для проверки метода upload VolunteerController и начал писать тесты для проверки метода updateName VolunteerController.

Manikgog committed:
Написаны все методы в классе VolunteerControllerTestRestTemplateTest для проверки работы VolunteerController, VolunteerService и VolunteerRepo.

Manikgog committed:
Сделал сущность отчёта и начал писать ReportController, ReportService и ReportRepo.

Manikgog committed:
Написаны методы контроллера ReportController, сервиса ReportService и репозитория ReportRepo.

Manikgog committed:
Написаны методы для обработки запросов пользователя бота на получения информации о приютах, животных с фотографиями и описанием.

Manikgog committed:
Написан метод для отображения информации по требованиям к усыновителю.

ASanderD committed:
Merge remote-tracking branch 'origin/Maksim_feature_volunteer_crud' into develop

# Conflicts:
#	src/main/java/ru/pet/nursery/web/service/AnimalService.java
#	src/main/java/ru/pet/nursery/web/validator/Validator.java

ASanderD committed:
Merge remote-tracking branch 'origin/Vethanin-feature-shelter' into develop

# Conflicts:
#	src/main/resources/application.properties

Commit:
Исправил не рабочую пагинацию в приютах и юзерах, написал тесты

Manikgog committed:
Написаны тесты для AnimalService, ReportService, ReportValidator. Перерисована схема таблиц базы данных.

Manikgog committed:
Merge branch 'Vetchanin-feature-users' into develop

Manikgog committed:
Merge branch 'Maksim_feature_volunteer_crud' into develop

# Conflicts:
#	src/main/java/ru/pet/nursery/repository/AnimalRepo.java
#	src/main/java/ru/pet/nursery/web/controller/ShelterController.java
#	src/main/java/ru/pet/nursery/web/service/AnimalService.java
#	src/main/java/ru/pet/nursery/web/validator/Validator.java
#	src/test/java/ru/pet/nursery/AnimalServiceMockTest.java
#	src/test/java/ru/pet/nursery/web/Constants.java
#	src/test/java/ru/pet/nursery/web/controller/AnimalControllerTestRestTemplateTest.java

Manikgog committed:
Внес правки, чтобы программа запускалась и правильно работала с таблицей животных.

Manikgog committed:
Написал отравку диеты, состояния здоровья, описание поведения в отчёт.

Commit:
Добавлен сервис усыновителей и шедулинг

Commit:
Добавил контроллер для усыновителей, переделал шедулинг, добавил поле в Report nextReportDate.
Посмотри пожалуйста, я добавил метод который дергает шедулинг в ReportSer.

Manikgog committed:
Написал отравку фотографии питомца в отчёт.

Manikgog committed:
В классе Animal в методах, где делалась проверка на User c id = 1, сделана проверка на user == null. Также исправлена неправильная работа метода постраничного вывода списка животных.

Manikgog committed:
Исправлены unit тесты: AnimalServiceMockTest, ReportServiceMockTest, ReportValidatorUnitTest, ValidatorUnitTest, VolunteerValidatorUnitTest.

Manikgog committed:
Удалены папки Action и Commands со всем содержимым.

ASanderD committed:
Merge remote-tracking branch 'origin/develop' into develop

# Conflicts:
#	src/main/java/ru/pet/nursery/web/service/AnimalService.java
#	src/main/java/ru/pet/nursery/web/validator/Validator.java

Manikgog committed:
Начал писать ReportControllerTestRestTemplateTest.

Manikgog committed:
Начал писать ReportControllerTestRestTemplateTest.

Manikgog committed:
Написано 5 методов для тестирования в классе ReportControllerTestRestTemplateTest.

Manikgog committed:
Написаны тесты на все методы класса ReportController в классе ReportControllerTestRestTemplateTest.

Manikgog committed:
Начал работу по логированию.

dmitry-bizin commented on src/main/java/ru/pet/nursery/data/MessageData.java:
```diff
@@ -0,0 +1,20 @@
+import java.util.Map;
+
+public class MessageData {
+    public static final Map<Long, String> chatId_reportStatus = new HashMap<>();
```
UPPER_CASE

dmitry-bizin commented on src/main/java/ru/pet/nursery/data/MessageData.java:
```diff
@@ -0,0 +1,20 @@
+        chatId_reportStatus.put(chatId, reportStatus);
+    }
+
+    public static void removeFromChatId_reportStatusMap(long chatId){
```
camelCase

dmitry-bizin commented on src/main/java/ru/pet/nursery/data/MessageData.java:
```diff
@@ -0,0 +1,20 @@
+public class MessageData {
+    public static final Map<Long, String> chatId_reportStatus = new HashMap<>();
+
+    public static void putToChatId_reportStatusMap(long chatId, String reportStatus){
```
camelCase

dmitry-bizin commented on src/main/java/ru/pet/nursery/entity/Report.java:
```diff
@@ -0,0 +1,52 @@
+    private User user;
+
+    @Column(columnDefinition="TEXT", name="path_to_foto")
+    private String path_to_foto;
```
camelCase

dmitry-bizin commented on src/main/java/ru/pet/nursery/entity/Report.java:
```diff
@@ -0,0 +1,52 @@
+    private boolean behaviourIsAccepted;
+
+    @Column(name = "all_items_is_accepted", nullable = false)
+    private boolean allItemsIsAccepted;
```
это поле лишнее, т.к. оно вычислимое из других - если все остальные true, значит allItemsWereAccepted

dmitry-bizin commented on src/main/java/ru/pet/nursery/data/Command.java:
```diff
@@ -0,0 +1,8 @@
+package ru.pet.nursery.data;
+
+public class Command {
```
лучше сделать enum-ом

dmitry-bizin commented on src/main/java/ru/pet/nursery/data/CallbackData.java:
```diff
@@ -0,0 +1,24 @@
+package ru.pet.nursery.data;
+
+public class CallbackData {
```
лучше сделать enum-ом

dmitry-bizin commented on src/main/java/ru/pet/nursery/handler/Handler.java:
```diff
@@ -1,44 +1,65 @@
 @Service
+@Slf4j
 public class Handler {
+    private final Logger logger = LoggerFactory.getLogger(Handler.class);
```
Зачем логгер, если используется Slf4j аннотация?

dmitry-bizin commented on src/main/java/ru/pet/nursery/data/MessageData.java:
```diff
@@ -0,0 +1,20 @@
+import java.util.HashMap;
+import java.util.Map;
+
+public class MessageData {
```
этот класс лучше сделать бином, статик методы и поля убрать вообще везде, вы на спринге пишете, поэтому надо играть по его правилам - создавать бины и использовать бины, а не статик методы и поля

dmitry-bizin commented on src/main/java/ru/pet/nursery/manager/info/InfoManager.java:
```diff
@@ -0,0 +1,376 @@
+    private final ShelterService shelterService;
+    private final AnimalService animalService;
+
+    private final Map<Long, Integer> userId_animalId = new HashMap<>(); // словарь соответствия chatId идентификатору животного, чтобы передавать новую фотографию
```
camelCase

dmitry-bizin commented on src/main/java/ru/pet/nursery/manager/report/ReportManager.java:
```diff
@@ -0,0 +1,435 @@
+
+@Component
+public class ReportManager extends AbstractManager {
+    private final String REPORT_PHOTO = "report_photo";
```
camelCase или static И final

dmitry-bizin commented on src/main/java/ru/pet/nursery/manager/report/ReportManager.java:
```diff
@@ -0,0 +1,435 @@
+            PhotoSize[] photos = update.message().photo();
+            Document document = update.message().document();
+            String fileId = null;
+            String extension = ".png";
+            if(photos != null){
+                fileId = photos[photos.length - 1].fileId();
+            } else if (document != null) {
+                fileId = document.fileId();
+                String mime = document.mimeType();
+                switch (mime){
+                    case "image/png" -> extension = ".png";
+                    case "image/jpeg" -> extension = ".jpg";
+                    default -> {
+                        sendMessage(adopterId, "Формат файла не поддерживается");
+                        return;
+                    }
+                }
+            }
+            String response = String.format(
```
это всё не нужно, надо просто взять название файла и при помощи метода getFilenameExtension у StringUtils спрингового в одну строку узнать расширение

dmitry-bizin commented on src/main/java/ru/pet/nursery/manager/report/ReportManager.java:
```diff
@@ -0,0 +1,435 @@
+                }
+            }
+            String response = String.format(
+                    "https://api.telegram.org/bot%s/getFile?file_id=%s",
+                    telegramBot.getToken(),
+                    fileId);
+
+            URL url = new URL(response);
+
+            String strPath = System.getProperty("user.dir");
+            if(strPath.contains("\\")){
+                strPath += "\\";
+            }else{
+                strPath += "/";
+            }
+            strPath += REPORT_PHOTO;
+            Path path = Path.of(strPath);
+            Path filePath = Path.of(path.toString(),  report.getId() + "_" + LocalDate.now() + extension);
+            Files.createDirectories(filePath.getParent());
+            Files.deleteIfExists(filePath);
```
это всё тоже не нужно, получить файл можно в 2-3 строки при помощи методов библиотеки, не надо руками ничего скачивать, библиотека умеет это делать сама.

dmitry-bizin replied:
слишком много лишнего кода, изобретение велосипеда, упростить можно раза в 4

dmitry-bizin replied:
(см. документацию библиотеки)

dmitry-bizin commented on src/main/java/ru/pet/nursery/repository/NurseryRepo.java:
```diff
@@ -0,0 +1,8 @@
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+import ru.pet.nursery.entity.Nursery;
+@Repository
```
аннотация не нужна

dmitry-bizin commented on src/main/java/ru/pet/nursery/repository/UserRepo.java:
```diff
@@ -0,0 +1,8 @@
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+import ru.pet.nursery.entity.User;
+@Repository
```
аннотация не нужна

dmitry-bizin commented on src/main/java/ru/pet/nursery/repository/VolunteerRepo.java:
```diff
@@ -3,7 +3,7 @@
 import org.springframework.stereotype.Repository;
 import ru.pet.nursery.entity.Volunteer;
-
 @Repository
```
аннотация не нужна

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/controller/AnimalController.java:
```diff
@@ -20,27 +29,188 @@ public class AnimalController {
+
+                    )
+            })
+    @DeleteMapping("/{id}/delete")
```
в прошлый раз писал уже, что никаких слов в путях быть не должно

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/controller/ReportController.java:
```diff
@@ -0,0 +1,276 @@
+                            )
+                    )
+            })
+    @GetMapping("/{date}")
```
дата должны идти как параметр запроса, а не пути (см. принципы REST)

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/controller/ShelterController.java:
```diff
@@ -176,10 +174,10 @@ public ResponseEntity<List<Nursery>> getNursery(@RequestParam @Min(1) int page,
-    public ResponseEntity<List<Nursery>> getShelterForKindOfAnimals(@RequestParam Boolean kindOfAnimal,
-                                                                    @RequestParam @Min(1) int page,
-                                                                    @RequestParam @Min(1) int size) {
+    @GetMapping("/getShelterForKindOfAnimals")
```
в прошлый раз писал уже, что никаких слов в путях быть не должно

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/controller/ShelterController.java:
```diff
@@ -150,8 +149,7 @@ public ResponseEntity<Nursery> deleteShelter(@PathVariable Long shelterId) {
     })
-    @GetMapping
-
+    @GetMapping(name = "/getNursery", params = {"page","size"})
```
в прошлый раз писал уже, что никаких слов в путях быть не должно

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/controller/VolunteerController.java:
```diff
@@ -0,0 +1,200 @@
+
+                    )
+            })
+    @PutMapping(value = "/{id}/{status}")
```
не должно быть в путях никаких параметров, кроме id - см. принципы REST

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/controller/VolunteerController.java:
```diff
@@ -0,0 +1,200 @@
+
+                    )
+            })
+    @PutMapping(value = "/{id}/name")
```
никаких слов тоже быть не должно

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/dto/AnimalDTO.java:
```diff
@@ -1,18 +1,51 @@
-    private String animalType;
-    private String gender;
+
+    @Enumerated(EnumType.STRING)
```
эти аннотации предназначены для сущностей и работать будут только там, здесь в dto они бесполезны и не имеют никакого смысла

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/dto/AnimalDTO.java:
```diff
@@ -1,18 +1,51 @@
+    @Enumerated(EnumType.STRING)
+    private AnimalType animalType;
+
+    @Enumerated(EnumType.STRING)
```
эти аннотации предназначены для сущностей и работать будут только там, здесь в dto они бесполезны и не имеют никакого смысла

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/dto/AnimalDTOForUser.java:
```diff
@@ -0,0 +1,47 @@
+
+    private String animalName;
+
+    @Enumerated(EnumType.STRING)
```
эти аннотации предназначены для сущностей и работать будут только там, здесь в dto они бесполезны и не имеют никакого смысла

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/dto/AnimalDTOForUser.java:
```diff
@@ -0,0 +1,47 @@
+    @Enumerated(EnumType.STRING)
+    private AnimalType animalType;
+
+    @Enumerated(EnumType.STRING)
```
эти аннотации предназначены для сущностей и работать будут только там, здесь в dto они бесполезны и не имеют никакого смысла

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/dto/AnimalDTOForUser.java:
```diff
@@ -0,0 +1,47 @@
+
+    private LocalDate birthDate;
+
+    private Nursery nursery;
```
в dto не должно быть сущностей. в dto должны быть только другие dto, иначе смысл dto теряется

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/exception/IllegalFieldException.java:
```diff
@@ -0,0 +1,18 @@
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(HttpStatus.BAD_REQUEST)
```
нет никакого смысла в этой аннотации, т.к. исключения перехватывается ВАМИ в handler-е и ВЫ САМИ там возвращаете нужный код ответа

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/exception/IllegalParameterException.java:
```diff
@@ -0,0 +1,18 @@
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(HttpStatus.BAD_REQUEST)
```
нет никакого смысла в этой аннотации, т.к. исключения перехватывается ВАМИ в handler-е и ВЫ САМИ там возвращаете нужный код ответа

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/exception/PageSizeException.java:
```diff
@@ -0,0 +1,18 @@
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(HttpStatus.BAD_REQUEST)
```
нет никакого смысла в этой аннотации, т.к. исключения перехватывается ВАМИ в handler-е и ВЫ САМИ там возвращаете нужный код ответа

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/exception/ReportIsExistException.java:
```diff
@@ -0,0 +1,18 @@
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(HttpStatus.BAD_REQUEST)
```
нет никакого смысла в этой аннотации, т.к. исключения перехватывается ВАМИ в handler-е и ВЫ САМИ там возвращаете нужный код ответа

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/exception/UserNotFoundException.java:
```diff
@@ -0,0 +1,11 @@
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(HttpStatus.NOT_FOUND)
```
нет никакого смысла в этой аннотации, т.к. исключения перехватывается ВАМИ в handler-е и ВЫ САМИ там возвращаете нужный код ответа

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/exception/UserNotValidException.java:
```diff
@@ -0,0 +1,11 @@
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(HttpStatus.BAD_REQUEST)
```
нет никакого смысла в этой аннотации, т.к. исключения перехватывается ВАМИ в handler-е и ВЫ САМИ там возвращаете нужный код ответа

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/AnimalService.java:
```diff
@@ -1,33 +1,58 @@
 
 @Service
 public class AnimalService {
+    private final Logger logger = LoggerFactory.getLogger(AbstractManager.class);
```
название класса неверное

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/AnimalService.java:
```diff
@@ -56,17 +85,22 @@ public ResponseEntity<Animal> uploadAnimal(AnimalDTO animalDTO) {
+            throw new EntityNotFoundException(animalId);
         }
         String strPath = System.getProperty("user.dir");
+        if(strPath.contains("\\")){
+            strPath += "\\";
+        }else{
+            strPath += "/";
+        }
```
не нужно никаких костылей, надо правильно использовать File API, оно само умеет правильно учитывать разделители

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/AnimalService.java:
```diff
@@ -124,4 +163,148 @@ public void getAnimalPhoto(int id, HttpServletResponse response) throws IOExcept
+     */
+    public ResponseEntity<List<Animal>> getAll() {
+        logger.info("Method getAll of AnimalService class");
+        return ResponseEntity.of(Optional.of(animalRepo.findAll()));
```
зачем здесь Optional? какой в этом смысл-то?

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/AnimalService.java:
```diff
@@ -124,4 +163,148 @@ public void getAnimalPhoto(int id, HttpServletResponse response) throws IOExcept
+        logger.info("Method getById of AnimalService class with parameter long -> {}", animalId);
+        AnimalDTOForUserMapper animalDTOForUserMapper = new AnimalDTOForUserMapper();
+        Animal animalFromDB = animalRepo.findById(animalId).orElseThrow(() -> new EntityNotFoundException(animalId));
+        return ResponseEntity.of(Optional.of(animalDTOForUserMapper.perform(animalFromDB)));
```
зачем здесь Optional? какой в этом смысл-то?

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/AnimalService.java:
```diff
@@ -124,4 +163,148 @@ public void getAnimalPhoto(int id, HttpServletResponse response) throws IOExcept
+        animalOld.setUser(null);
+        animalOld.setTookDate(null);
+        Animal newAnimal = animalRepo.save(animalOld);
+        return ResponseEntity.of(Optional.of(newAnimal));
```
зачем здесь Optional? какой в этом смысл-то?

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/AnimalService.java:
```diff
@@ -124,4 +163,148 @@ public void getAnimalPhoto(int id, HttpServletResponse response) throws IOExcept
+        PageRequest pageRequest = PageRequest.of(pageNumber - 1, pageSize);
+        List<Animal> animals = animalRepo.findByUserIsNull(pageRequest);
+        List<AnimalDTOForUser> resultAnimals = convertListAnimalToListAnimalDTO(animals);
+        return ResponseEntity.of(Optional.of(resultAnimals));
```
зачем здесь Optional? какой в этом смысл-то?

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/AnimalService.java:
```diff
@@ -124,4 +163,148 @@ public void getAnimalPhoto(int id, HttpServletResponse response) throws IOExcept
+        animalFromDB.setUser(userAdopted);
+        animalFromDB.setTookDate(LocalDate.now());
+        Animal newAnimal = animalRepo.save(animalFromDB);
+        return ResponseEntity.of(Optional.of(newAnimal));
```
зачем здесь Optional? какой в этом смысл-то?

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/AnimalService.java:
```diff
@@ -37,14 +62,18 @@ public AnimalService(AnimalRepo animalRepo){
+        newAnimal.setNursery(nursery);
+        newAnimal.setUser(null);                            // если стоит цифра 1 значит животное никто не взял
         Animal animalFromDB = animalRepo.save(newAnimal);
         return ResponseEntity.of(Optional.of(animalFromDB));
```
зачем здесь Optional? какой в этом смысл-то?

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/ReportService.java:
```diff
@@ -0,0 +1,271 @@
+    public ResponseEntity<Report> delete(long id) {
+        Report reportFromDB = reportRepo.findById(id).orElseThrow(() -> new EntityNotFoundException(id));
+        reportRepo.delete(reportFromDB);
+        return ResponseEntity.of(Optional.of(reportFromDB));
```
зачем здесь Optional? какой в этом смысл-то?

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/ReportService.java:
```diff
@@ -0,0 +1,271 @@
+     * @return ResponseEntity.ok()
+     * @throws IOException - исключение ввода-вывода
+     */
+    public ResponseEntity updateFoto(long id, MultipartFile reportFoto) throws IOException {
```
```suggestion
    public ResponseEntity<?> updateFoto(long id, MultipartFile reportFoto) throws IOException {
```

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/ReportService.java:
```diff
@@ -0,0 +1,271 @@
+        validator.stringValidate(diet);
+        reportOld.setDiet(diet);
+        Report reportNew = reportRepo.save(reportOld);
+        return ResponseEntity.of(Optional.of(reportNew));
```
зачем здесь Optional? какой в этом смысл-то?

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/ReportService.java:
```diff
@@ -0,0 +1,271 @@
+
+    private void updateFotoPathColumn(String path, long id){
+        Report reportOld = reportRepo.findById(id).orElseThrow(() -> new EntityNotFoundException(id));
+        reportOld.setPath_to_foto(path);
```
camelCase

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/ReportService.java:
```diff
@@ -0,0 +1,271 @@
+        validator.stringValidate(health);
+        reportOld.setHealth(health);
+        Report reportNew = reportRepo.save(reportOld);
+        return ResponseEntity.of(Optional.of(reportNew));
```
зачем здесь Optional? какой в этом смысл-то?

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/ReportService.java:
```diff
@@ -0,0 +1,271 @@
+        reportOld.setBehaviour(behaviour);
+        Report reportNew = reportRepo.save(reportOld);
+
+        return ResponseEntity.of(Optional.of(reportNew));
```
зачем здесь Optional? какой в этом смысл-то?

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/ReportService.java:
```diff
@@ -0,0 +1,271 @@
+        Report reportOld = reportRepo.findById(id).orElseThrow(() -> new EntityNotFoundException(id));
+        reportOld.setAllItemsIsAccepted(isAllItemsAccepted);
+        Report reportNew = reportRepo.save(reportOld);
+        return ResponseEntity.of(Optional.of(reportNew));
```
зачем здесь Optional? какой в этом смысл-то?

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/ReportService.java:
```diff
@@ -0,0 +1,271 @@
+        Report reportOld = reportRepo.findById(id).orElseThrow(() -> new EntityNotFoundException(id));
+        reportOld.setPhotoIsAccepted(isPhotoAccepted);
+        Report reportNew = reportRepo.save(reportOld);
+        return ResponseEntity.of(Optional.of(reportNew));
```
зачем здесь Optional? какой в этом смысл-то?

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/ReportService.java:
```diff
@@ -0,0 +1,271 @@
+        Report reportOld = reportRepo.findById(id).orElseThrow(() -> new EntityNotFoundException(id));
+        reportOld.setDietIsAccepted(isDietAccepted);
+        Report reportNew = reportRepo.save(reportOld);
+        return ResponseEntity.of(Optional.of(reportNew));
```
зачем здесь Optional? какой в этом смысл-то?

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/ReportService.java:
```diff
@@ -0,0 +1,271 @@
+        Report reportOld = reportRepo.findById(id).orElseThrow(() -> new EntityNotFoundException(id));
+        reportOld.setHealthIsAccepted(isHealthAccepted);
+        Report reportNew = reportRepo.save(reportOld);
+        return ResponseEntity.of(Optional.of(reportNew));
```
зачем здесь Optional? какой в этом смысл-то?

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/ReportService.java:
```diff
@@ -0,0 +1,271 @@
+        Report reportOld = reportRepo.findById(id).orElseThrow(() -> new EntityNotFoundException(id));
+        reportOld.setBehaviourIsAccepted(isBehaviourAccepted);
+        Report reportNew = reportRepo.save(reportOld);
+        return ResponseEntity.of(Optional.of(reportNew));
```
зачем здесь Optional? какой в этом смысл-то?

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/ReportService.java:
```diff
@@ -0,0 +1,271 @@
+     * @return ResponseEntity<List<Report>> - список объектов Report
+     */
+    public ResponseEntity<List<Report>> getListOfReportByDate(LocalDate date) {
+        return ResponseEntity.of(Optional.ofNullable(reportRepo.findByReportDate(date)));
```
зачем здесь Optional? какой в этом смысл-то?

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/ReportService.java:
```diff
@@ -0,0 +1,271 @@
+        newReport.setUser(user);
+        newReport.setReportDate(LocalDate.now());
+        Report reportFromDB = reportRepo.save(newReport);
+        return ResponseEntity.of(Optional.of(reportFromDB));
```
зачем здесь Optional? какой в этом смысл-то?

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/VolunteerService.java:
```diff
@@ -0,0 +1,144 @@
+        volunteer.setId(0);
+        validator.validate(volunteer);
+        Volunteer volunteerFromDB = volunteerRepo.save(volunteer);
+        return ResponseEntity.of(Optional.of(volunteerFromDB));
```
зачем здесь Optional? какой в этом смысл-то?

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/VolunteerService.java:
```diff
@@ -0,0 +1,144 @@
+     * @return ResponseEntity.of(Optional.of(volunteerFromDB))
+     */
+    public ResponseEntity<Volunteer> upload(Volunteer volunteer) {
+        volunteer.setId(0);
```
зачем? id не надо присваивать руками, этим занимается БД

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/VolunteerService.java:
```diff
@@ -0,0 +1,144 @@
+        validator.validateId(id);
+        Volunteer volunteerOld = volunteerRepo.findById(id).get();
+        if(volunteerOld.getName().equals(name)){
+            return ResponseEntity.of(Optional.of(volunteerOld));
```
зачем здесь Optional? какой в этом смысл-то?

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/VolunteerService.java:
```diff
@@ -0,0 +1,144 @@
+        }
+        volunteerOld.setName(name);
+        Volunteer volunteer = volunteerRepo.save(volunteerOld);
+        return ResponseEntity.of(Optional.of(volunteer));
```
зачем здесь Optional? какой в этом смысл-то?

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/VolunteerService.java:
```diff
@@ -0,0 +1,144 @@
+        }
+        volunteerOld.setActive(status);
+        Volunteer volunteer = volunteerRepo.save(volunteerOld);
+        return ResponseEntity.of(Optional.of(volunteer));
```
зачем здесь Optional? какой в этом смысл-то?

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/VolunteerService.java:
```diff
@@ -0,0 +1,144 @@
+        validator.validateId(id);
+        Volunteer volunteerOld = volunteerRepo.findById(id).get();
+        if(volunteerOld.getName().equals(phone)){
+            return ResponseEntity.of(Optional.of(volunteerOld));
```
зачем здесь Optional? какой в этом смысл-то?

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/VolunteerService.java:
```diff
@@ -0,0 +1,144 @@
+        }
+        volunteerOld.setPhoneNumber(phone);
+        Volunteer volunteer = volunteerRepo.save(volunteerOld);
+        return ResponseEntity.of(Optional.of(volunteer));
```
зачем здесь Optional? какой в этом смысл-то?

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/VolunteerService.java:
```diff
@@ -0,0 +1,144 @@
+     */
+    public ResponseEntity<List<Volunteer>> getAll() {
+        List<Volunteer> volunteers = volunteerRepo.findAll();
+        return ResponseEntity.of(Optional.of(volunteers));
```
зачем здесь Optional? какой в этом смысл-то?

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/VolunteerService.java:
```diff
@@ -0,0 +1,144 @@
+        validator.validateId(id);
+        Optional<Volunteer> volunteer = volunteerRepo.findById(id);
+        volunteerRepo.delete(volunteer.get());
+        return ResponseEntity.of(volunteer);
```
зачем здесь Optional? какой в этом смысл-то?

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/VolunteerService.java:
```diff
@@ -0,0 +1,144 @@
+     */
+    public ResponseEntity<Volunteer> get(Integer id) {
+        validator.validateId(id);
+        return ResponseEntity.of(volunteerRepo.findById(id));
```
зачем здесь Optional? какой в этом смысл-то?

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/VolunteerService.java:
```diff
@@ -0,0 +1,144 @@
+        volunteerOld.setName(volunteer.getName());
+        volunteerOld.setPhoneNumber(volunteer.getPhoneNumber());
+        Volunteer volunteerNew = volunteerRepo.save(volunteerOld);
+        return ResponseEntity.of(Optional.of(volunteerNew));
```
зачем здесь Optional? какой в этом смысл-то?

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/VolunteerService.java:
```diff
@@ -0,0 +1,144 @@
+        validator.validate(volunteer);
+        Volunteer volunteerOld = volunteerRepo.findById(volunteer.getId()).get();
+        if(volunteerOld.equals(volunteer)){
+            return ResponseEntity.of(Optional.of(volunteerOld));
```
зачем здесь Optional? какой в этом смысл-то?

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/VolunteerService.java:
```diff
@@ -0,0 +1,144 @@
+    public ResponseEntity<Volunteer> updateName(String name, int id) {
+        validator.stringValidate(name);
+        validator.validateId(id);
+        Volunteer volunteerOld = volunteerRepo.findById(id).get();
```
неправильная работа с Optional - get вызывается без проверки, будет исключение

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/VolunteerService.java:
```diff
@@ -0,0 +1,144 @@
+     */
+    public ResponseEntity<Volunteer> updateStatus(Boolean status, Integer id) {
+        validator.validateId(id);
+        Volunteer volunteerOld = volunteerRepo.findById(id).get();
```
неправильная работа с Optional - get вызывается без проверки, будет исключение

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/VolunteerService.java:
```diff
@@ -0,0 +1,144 @@
+        validator.stringValidate(phone);
+        validator.phoneValidate(phone);
+        validator.validateId(id);
+        Volunteer volunteerOld = volunteerRepo.findById(id).get();
```
неправильная работа с Optional - get вызывается без проверки, будет исключение

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/VolunteerService.java:
```diff
@@ -0,0 +1,144 @@
+    public ResponseEntity<Volunteer> updateVolunteer(Integer id, Volunteer volunteer) {
+        validator.validateId(id);
+        validator.validate(volunteer);
+        Volunteer volunteerOld = volunteerRepo.findById(volunteer.getId()).get();
```
неправильная работа с Optional - get вызывается без проверки, будет исключение

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/validator/Validator.java:
```diff
@@ -0,0 +1,159 @@
+     * @param pageSize - целое число
+     */
+    public void validatePageSize(Integer pageSize) {
+        if(pageSize <= 0)
```
скобки надо писать всегда

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/validator/Validator.java:
```diff
@@ -0,0 +1,159 @@
+     * @param pageNumber - целое число
+     */
+    public void validatePageNumber(Integer pageNumber) {
+        if(pageNumber <= 0)
```
скобки надо писать всегда

dmitry-bizin commented on src/main/resources/application.properties:
```diff
@@ -1,8 +1,9 @@
-spring.datasource.password=12345678
+spring.datasource.username=${USER_NAME}
+spring.datasource.password=${DATASOURCE_PASSWORD}
 spring.jpa.hibernate.ddl-auto=update
```
стратегия должны быть validate, писал в прошлый раз уже

Manikgog committed:
Начал писать создание таблиц с помощью liquibase.

Manikgog committed:
Написал создание таблиц liquibase и исправил баги, которые возникали при обращении из бота если списки животных и питомников пустые.

Manikgog committed:
Исправил тесты, чтобы они работали с liquibase.

Manikgog committed:
Убрал лишние строки.

Manikgog committed:
Покрытие тестами AnimalController, AnimalService, ReportController, ReportService, VolunteerController, VolunteerService доведено до 100%.

Manikgog committed:
Добавлены комментарии к методам класса ReportManager. Обработано исключение IOException, которое выбрасывает метод uploadPhotoToReport класса ReportManager в методе класса ReportHandler answer.

Manikgog committed:
Добавлены комментарии к методам класса ReportManager. Обработано исключение IOException, которое выбрасывает метод uploadPhotoToReport класса ReportManager в методе класса ReportHandler answer.

Manikgog committed:
Merge branch 'Vetchanin-feature-usersTG' into develop

# Conflicts:
#	src/main/java/ru/pet/nursery/commands/Commands.java
#	src/main/java/ru/pet/nursery/entity/Animal.java
#	src/main/java/ru/pet/nursery/entity/Report.java
#	src/main/java/ru/pet/nursery/manager/info/InfoManager.java
#	src/main/java/ru/pet/nursery/mapper/AnimalMapper.java
#	src/main/java/ru/pet/nursery/repository/AnimalRepo.java
#	src/main/java/ru/pet/nursery/web/service/AnimalService.java
#	src/main/java/ru/pet/nursery/web/service/ReportService.java

Manikgog committed:
Исправлены методы для использования LocalDateTime.

Manikgog committed:
В класс Handler добавлено действие по добавлению пользователя в базу данных, если его там не было.

Manikgog committed:
В класс Handler добавлено действие по добавлению пользователя в базу данных, если его там не было.

Manikgog committed:
В класс Handler добавлено действие по добавлению пользователя в базу данных, если его там не было.

Commit:
Дописал тесты

Commit:
И еще не много тестов

Manikgog committed:
Добавлены интерфейсы для сервисов.

Manikgog committed:
Merge remote-tracking branch 'origin/Vetchanin-tests' into Vetchanin-tests

Manikgog committed:
Добавлены интерфейсы для сервисов.

Commit:
И еще немного тестов

Commit:
Merge remote-tracking branch 'origin/Vetchanin-tests' into Vetchanin-tests

Commit:
Изменение в нотифаер

Manikgog committed:
Написаны тесты на все метода класса InfoManager.

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/controller/AdoptedController.java:
```diff
@@ -0,0 +1,32 @@
+    }
+
+    @PostMapping
+    public ResponseEntity<Animal> setAdopterForAnimal(@PathVariable Long animalId, @PathVariable Long adopterId) {
```
эндпоинт нерабочий - нет параметра пути в пути

dmitry-bizin replied:
`/{adopterId}?animalId=`
вот так лучше, т.е. в пути только id адоптера

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/controller/AdoptedController.java:
```diff
@@ -0,0 +1,32 @@
+        return ResponseEntity.ok(adoptedService.setAdopterForAnimal(animalId, adopterId));
+    }
+    @PutMapping
+    public ResponseEntity<Animal> prolongTrialForNDays(@PathVariable Long animalId, @PathVariable Integer days) {
```
эндпоинт нерабочий - нет параметра пути в пути

dmitry-bizin replied:
параметры лучше сделать параметрами запроса (RequestParam), а не пути

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/controller/AdoptedController.java:
```diff
@@ -0,0 +1,32 @@
+    }
+
+    @PutMapping("/cancelTrial")
+    public ResponseEntity<Animal> cancelTrial(Long animalId) {
```
нужно поставить аннотацию RequestParam

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/exception/AnimalNotFoundException.java:
```diff
@@ -0,0 +1,11 @@
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(HttpStatus.NOT_FOUND)
```
аннотация не нужна, она всё равно не применяется, потому что исключения вы сами ловите через хендлер и там же сами указываете код ответа

Manikgog committed:
Написаны тесты методов класса ReportManager.

Manikgog committed:
Написаны тесты методов класса StartManager и VolunteerManager.

Manikgog committed:
Merge branch 'refs/heads/Maksim_writing_tests' into develop

# Conflicts:
#	src/main/java/ru/pet/nursery/handler/Handler.java
#	src/main/java/ru/pet/nursery/web/service/AnimalService.java
#	src/main/java/ru/pet/nursery/web/service/ReportService.java
#	src/test/java/ru/pet/nursery/web/service/ReportServiceMockTest.java

Manikgog committed:
Написаны тесты для контроллеров, сервисов и менеджеров.

Manikgog committed:
Типы полей таблицы animal_table took_date pet_return_date исправлены с TIMESTAMP на Date.

Manikgog committed:
Исправлены ошибки в сущности User в методе getUserFullName, а также ошибка при сохранении отчёта. Т.к. LocalDateTime учитывает милисекунды, то
при поиске отчёта по дате отчёт не находился.

Manikgog committed:
Начал писать тест TelegramBotUpdatesListenerTest.

dmitry-bizin commented on src/main/java/ru/pet/nursery/handler/Handler.java:
```diff
@@ -1,44 +1,94 @@
+    public void answer(Update update) throws IOException {
+        logger.info("Processing update in method answer of Handler class: {}", update);
+        String updateStr = new Gson().toJson(update);
+        System.out.println(updateStr);
```
логгер для вывода используем всегда

dmitry-bizin commented on src/main/java/ru/pet/nursery/manager/info/InfoManager.java:
```diff
@@ -0,0 +1,426 @@
+    private final IShelterService shelterService;
+    private final AnimalRepo animalRepo;
+    private final IAnimalService animalService;
+    private final Map<Long, Long> userId_animalId = new HashMap<>(); // словарь соответствия chatId идентификатору животного, чтобы передавать новую фотографию
```
camelCase, а не snake_case

dmitry-bizin commented on src/main/java/ru/pet/nursery/manager/info/InfoManager.java:
```diff
@@ -0,0 +1,426 @@
+     */
+    public void dogPhoto(CallbackQuery callbackQuery) throws IOException {
+        String cbq = new Gson().toJson(callbackQuery);
+        System.out.println(cbq);
```
logger

dmitry-bizin commented on src/main/java/ru/pet/nursery/manager/info/InfoManager.java:
```diff
@@ -0,0 +1,426 @@
+     * @throws IOException - исключение ввода-вывода
+     */
+    public void dogPhoto(CallbackQuery callbackQuery) throws IOException {
+        String cbq = new Gson().toJson(callbackQuery);
```
ObjectMapper

Manikgog committed:
Удалены HttpStatus'ы из всех классов исключений, т.к. они исключения обрабатываются в NurseryExceptionHandler и там указывается код ответа.
Для определения размера файла в методе getAnimalPhoto класса AnimalService применён метод Files.readAllBytes(path).length.
Для возвращения файла в методе getPhotoByteArray класса AnimalService применён метод Files.readAllBytes(path).

MNGogolin committed:
Исправлены тесты, которые проверяли методы по выводу фотографий питомцев.

MNGogolin committed:
Внес действие бота при нераспознанном сообщении от пользователя телеграм.

MNGogolin committed:
Добавлен класс для тестирования TelegramBotUpdatesListenerIntegratedTest.

MNGogolin committed:
Добавлен класс для тестирования TelegramBotUpdatesListenerIntegratedTest. Убраны два метода dogsInformation и catsInformation из класса InfoManager и две строки из метода answer класса CallbackQueryHandler из switch case, которые вызывали эти метод

Manikgog committed:
Написан тест для класса TelegramBotUpdatesListener.

Manikgog committed:
Merge branch 'refs/heads/Maksim_fix_bug_in_infoManager' into develop

Commit:
Изменение в нотифаер, поправил эндопинты
в AdoptedController исправил тесты к этому же контроллеру

Commit:
Добавил логгеры в сревисы, добавил напоминание о завершении адаптационного периода и отправка напоминания пользователю

Commit:
Фикс нотифаер

Manikgog committed:
Добавлена возможность посмотреть где находятся приюты перейдя по ссылке.

Manikgog committed:
Merge branch 'refs/heads/Vetchanin-tests' into develop

# Conflicts:
#	src/main/java/ru/pet/nursery/repository/ReportRepo.java
#	src/main/java/ru/pet/nursery/web/service/AdoptedService.java
#	src/main/java/ru/pet/nursery/web/service/IAnimalService.java
#	src/main/java/ru/pet/nursery/web/service/IShelterService.java
#	src/main/java/ru/pet/nursery/web/service/ReportService.java
#	src/main/java/ru/pet/nursery/web/service/ShelterService.java
#	src/test/java/ru/pet/nursery/web/ReportValidatorUnitTest.java
#	src/test/java/ru/pet/nursery/web/controller/AnimalControllerTestRestTemplateTest.java
#	src/test/java/ru/pet/nursery/web/controller/ReportControllerTestRestTemplateTest.java
#	src/test/java/ru/pet/nursery/web/service/ShelterServiceTest.java
#	src/test/java/ru/pet/nursery/web/service/UserServiceTest.java

Manikgog committed:
Исправлены ошибки.

Manikgog committed:
Исправлены ошибки.

Commit:
Фикс тестов AdoptedControllerTestTestRestTemplateTest, AdoptedServiceTest, ShelterServiceTest, AdoptedService

Manikgog committed:
Merge branch 'refs/heads/Vetchanin-added-logger' into develop

# Conflicts:
#	src/main/java/ru/pet/nursery/notifying/Notifier.java
#	src/main/java/ru/pet/nursery/web/controller/UserController.java
#	src/main/java/ru/pet/nursery/web/service/AdoptedService.java
#	src/main/java/ru/pet/nursery/web/service/ShelterService.java
#	src/main/java/ru/pet/nursery/web/service/UserService.java
#	src/test/java/ru/pet/nursery/web/service/UserServiceTest.java

Manikgog committed:
Исправлены ошибки в тестах InfoManagerMockTest.

Manikgog committed:
Merge remote-tracking branch 'refs/remotes/origin/Vetchanin-fixed-tetst-ShelterServiceTest' into develop

# Conflicts:
#	src/main/java/ru/pet/nursery/web/service/AdoptedService.java
#	src/test/java/ru/pet/nursery/web/service/AdoptedServiceTest.java

Manikgog committed:
Исправлены ошибки в тестах.

ASanderD committed:
Добавлена связь пользователь-волонтер.

Manikgog committed:
Merge branch 'refs/heads/adubograev_connection_User_Volunteer' into develop

Manikgog committed:
Merge branch 'refs/heads/adubograev_connection_User_Volunteer' into develop

# Conflicts:
#	src/main/java/ru/pet/nursery/entity/User.java
#	src/main/java/ru/pet/nursery/handler/CallbackQueryHandler.java
#	src/main/java/ru/pet/nursery/manager/volunteer/VolunteerManager.java
#	src/main/java/ru/pet/nursery/web/service/UserService.java

dmitry-bizin commented on src/main/java/ru/pet/nursery/manager/info/InfoManager.java:
```diff
@@ -0,0 +1,378 @@
+            );
+            return;
+        }
+        long id = cats.get((int)getNextId(chatId, cats)).getId();
```
а почему у вас индекс - это id? правильнее сделать мапу id на животное

dmitry-bizin commented on src/main/java/ru/pet/nursery/manager/info/InfoManager.java:
```diff
@@ -0,0 +1,378 @@
+            );
+            return;
+        }
+        long id = dogs.get((int)getNextId(chatId, dogs)).getId();
```
а почему у вас индекс - это id? правильнее сделать мапу id на животное

dmitry-bizin commented on src/main/java/ru/pet/nursery/manager/info/InfoManager.java:
```diff
@@ -0,0 +1,378 @@
+     * @param animals - список животных
+     * @return новый идентификатор
+     */
+    public long getNextId(long chatId, List<Animal> animals){
```
костыльный метод, зачем вообще он нужен и почему именно так сделано? нет никакой гарантии, что id в БД будет вести себя так же, как вы здесь его увеличиваете. Мало ли в БД поменяется генерация id, тогда весь ваш код просто поломается

dmitry-bizin commented on src/main/java/ru/pet/nursery/notifying/Notifier.java:
```diff
@@ -0,0 +1,59 @@
+                    "У вас заканчивается адаптационный период "+animal.getPetReturnDate()+" у питомца "+animal.getAnimalName()+" " +
+                            ", пожалуйста свяжитесь с волонтером для продления срока пребывания");
+            telegramBot.execute(message);
+            List<Volunteer> volunteerList = volunteerService.getAll().stream().filter(Volunteer::isActive).toList();
```
не нужно никогда брать из БД всех, а затем в коде фильтровать. Надо сразу из БД брать нужных, чтобы меньше данных вытягивать и меньше заполнять хип лишними объектами сущностей

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/controller/ReportController.java:
```diff
@@ -0,0 +1,296 @@
+            }
+    )
+    @GetMapping("/{id}/photo")
+    public void getPhotoById(@PathVariable long id, HttpServletResponse response){
+        reportService.getPhotoById(id, response);
+    }
```
почему Get метод ничего не возвращает?

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/IReportService.java:
```diff
@@ -0,0 +1,4 @@
+package ru.pet.nursery.web.service;
+
+public interface IReportService {
```
зачем пустой интерфейс?

Manikgog replied:
Потому что моки без него не работают

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/IUserService.java:
```diff
@@ -0,0 +1,4 @@
+package ru.pet.nursery.web.service;
+
+public interface IUserService {
```
зачем пустой интерфейс?

Manikgog replied:
Потому что моки без него не работают

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/IVolunteerService.java:
```diff
@@ -0,0 +1,4 @@
+package ru.pet.nursery.web.service;
+
+public interface IVolunteerService {
```
зачем пустой интерфейс?

Manikgog replied:
Потому что моки без него не работают

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/ReportService.java:
```diff
@@ -0,0 +1,306 @@
+@Service
+public class ReportService implements IReportService {
+    private final Logger logger = LoggerFactory.getLogger(ReportService.class);
+    private final String REPORT_PHOTO = "test_report_photo";
```
в UPPER_CASE надо писать ТОЛЬКО поля, которые static и final

dmitry-bizin commented on src/main/java/ru/pet/nursery/web/service/ReportService.java:
```diff
@@ -0,0 +1,306 @@
+            return reportRepo.findByUserAndReportDate(user, LocalDateTime.now().truncatedTo(ChronoUnit.DAYS));
+        }
+        Report newReport = new Report();
+        newReport.setId(0);
```
зачем это? id должна присваивать ТОЛЬКО БД

Manikgog committed:
Исправлены замечения наставника.

Manikgog committed:
Исправлены замечения наставника.

Manikgog committed:
Исправлены замечения наставника.

Manikgog committed:
Из таблицы удален столбец all_items_is_accepted. Из сущности Report удалено поле allItemsIsAcceped. Удален метод контроллера и сервиса, связанные с этим полем.

Manikgog committed:
Удалены тесты для проверки методов, свзязанных с удаленным полем.

Manikgog committed:
Исправлены тесты.

Commit:
Фикс тестов и сервиса AdoptedService, UserControllerTestRestTemplateTest, UserServiceTest

Commit:
Переделал Notifier, VolunteerControllerTestRestTemplateTest, VolunteerRepo, VolunteerService

dmitry-bizin commented on src/main/java/ru/pet/nursery/data/ReportStatus.java:
```diff
@@ -0,0 +1,8 @@
+package ru.pet.nursery.data;
+
+public class ReportStatus {
```
лучше было бы сделать enum-ом статусы

dmitry-bizin commented on src/main/java/ru/pet/nursery/manager/report/ReportManager.java:
```diff
@@ -325,49 +329,31 @@ public void uploadPhotoToReport(Update update) throws IOException {
+            String urlTelegramFile = String.format("https://api.telegram.org/file/bot%s/%s",
                     telegramBot.getToken(),
-                    urlTelegramFile);
+                    file.filePath());
```
```java
byte[] image = telegramBot.getFileContent(file);
```
и всё, в одну строку

Manikgog committed:
Упрощены методы AnimalService, ReportService и ReportManager для записи фотографии животного.

Manikgog committed:
Удалена папка с классом Commands.

Manikgog committed:
Исправлены тесты. Добавлен файл application-prod.properties, который нужен для запуска приложения при готовом файле запуска.

Manikgog committed:
Добавлена схема таблиц базы данных.

ASanderD committed:
Переделан класс StartManager, Handler.
Написаны тесты для StartManager.
Удалены ненужные методы из UserService.

ASanderD committed:
Переделан класс StartManager, Handler.
Написаны тесты для StartManager.
Удалены ненужные методы из UserService.

Manikgog committed:
Merge branch 'refs/heads/Fixed-tests-and-services' into develop

# Conflicts:
#	src/test/java/ru/pet/nursery/web/controller/VolunteerControllerTestRestTemplateTest.java

Manikgog committed:
Дописан метод run() в notifier.

Commit:
BagFix Notifier

Manikgog committed:
Исправлен метод run() в notifier.
